src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,779 @@
+                    f"'{name}:{version}' in the MLflow model registry: {str(e)}"
+                )
+        return self.get_model_version(name, version)
+
+    def get_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> ModelVersion:
+        """Get a model version from the MLflow model registry.
+
+        Args:
+            name: The name of the model.
+            version: The version of the model.
+
+        Raises:
+            RuntimeError: If mlflow fails to get the model version.
+
+        Returns:
+            The model version.
+        """
+        # Get the model version from the MLflow model registry.
+        try:
+            mlflow_model_version = self.mlflow_client.get_model_version(
+                name=name,
+                version=version,
+            )
+        except MlflowException as e:
+            raise RuntimeError(
+                f"Failed to get model version '{name}:{version}' from the "
+                f"MLflow model registry: {str(e)}"
+            )
+        # Return the model version.
+        return self._cast_mlflow_version_to_model_version(
+            mlflow_model_version=mlflow_model_version,
+        )
+
+    def list_model_versions(
+        self,
+        name: Optional[str] = None,
+        model_source_uri: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
+    ) -> List[ModelVersion]:
+        """List model versions from the MLflow model registry.
+
+        Args:
+            name: The name of the model.
+            model_source_uri: The model source URI.
+            tags: The tags of the model version.
+            kwargs: Additional keyword arguments.
+
+        Returns:
+            The model versions.
+        """
+        # Set the filter string.
+        filter_string = ""
+        if name:
+            filter_string += f"name='{name}'"
+        if model_source_uri:
+            if filter_string:
+                filter_string += " AND "
+            filter_string += f"source='{model_source_uri}'"
+        if "mlflow_run_id" in kwargs and kwargs["mlflow_run_id"]:
+            if filter_string:
+                filter_string += " AND "
+            filter_string += f"run_id='{kwargs['mlflow_run_id']}'"
+        if tags:
+            for tag, value in tags.items():
+                if filter_string:
+                    filter_string += " AND "
+                filter_string += f"tags.{tag}='{value}'"
+        # Get the model versions.
+        mlflow_model_versions = self.mlflow_client.search_model_versions(
+            filter_string=filter_string,
+        )
+        # Return the model versions.
+        return [
+            self._cast_mlflow_version_to_model_version(
+                mlflow_model_version=mlflow_model_version,
+            )
+            for mlflow_model_version in mlflow_model_versions
+        ]
+
+    def get_latest_model_versions(
```

Comment:
This method is not in the model registry abstraction, yet I see you still call it from both seldon and mlflow model deployer steps.

Functionally, this works, but from a design perspective, in the seldon deployer, you should only use the abstraction methods, because otherwise your implementation isn't portable to another model registry. You should either make this method part of the model registry abstraction, or find some other method in the abstraction to achieve the same goal.

To me, this looks like what a more flexible `list_model_versions` method might do.